### PR TITLE
feat: Sort conversations by updated_at desc

### LINKED ISSUE #7338: Add filter by update time of conversations
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Our operations will check the logs daily to optimize our app, but the log page in our console cannot filter which session's update date (i.e. the time of the latest message for that session) can only be filtered based on the session's creation date, and cannot be filtered based on the session's update date. 
We hope to add the function of filtering based on update dates and add a column of update dates. This will help our operations to efficiently conduct log checks.

### 2. Additional context or comments

I found that in the conversation , the value in updated_at field always  equals the value in created_at field . 

### 3. Can you help us with this feature?

- [x] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by KinWang130**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] I have performed a self-review of my own code
- [x] I ran `dev/pr`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description
In the process of using dify for conversations, the historical conversation records are always sorted in descending order according to the time of session creation. Normally, they should be sorted according to the latest time of conversation messages.
My optimization is to modify the updated_at of the conversation during the conversation, and sort the conversation list in descending order based on the updated_at. This way, users can see the latest list of session messages.

Fixes https://github.com/langgenius/dify/issues/7338

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Initiate a new conversation, with the new conversation at the forefront
- [x] The conversation list is sorted based on the latest time of the conversation content

### CODE REVIEW

**Comment by laipz8200** [comment]
I think sorting by creation time is reasonable because it ensures the relative order of items in the list remains unchanged. Do you have a more compelling argument for sorting by update time?

**Comment by laipz8200** [comment]
> > I think sorting by creation time is reasonable because it ensures the relative order of items in the list remains unchanged. Do you have a more compelling argument for sorting by update time?
> 
> Of course, sorting by creation time can ensure that the relative order remains unchanged, but in most conversational applications such as GPT, WeChat, Telegram, etc., conversations are always sorted by the latest conversation messages. My optimization reasons are as follows:
> 
>     1. Updated_at is always unchanged. When a new message is generated in a conversation or the conversation name is modified, updated_at should be updated, otherwise it is meaningless
> 
>     2. Each conversation may generate a theme, and when users generate new messages in each conversation, I believe they want the conversation to be prioritized, which is very user-friendly
>        What's your opinion?

I think it would be great to support both options. How about we add a `sort_by` parameter to the API? That way, we could pass `sort_by=-updated_at` to sort by updated_at in descending order.

**Comment by leslie2046** [comment]
+1 
https://github.com/langgenius/dify/issues/7338
I think the conversation list on the left side of the webui should be sorted by 'updated_at'
And the log list in the console should also be sorted by updated_at. When we check the logs every day, we hope to only look at the current day's logs without having to repeatedly review the sessions we have already seen yesterday.

**Comment by laipz8200** [comment]
Hi @KinWang130! Apologies for the lack of clarity earlier. We would like to support the option to specify the sorting method in the web service API. Instead of directly changing the current sorting method, we aim to allow users to choose their preferred sorting method by adding a button in the frontend. Would you be willing to implement this feature?

**Comment by laipz8200** [comment]
@KinWang130 Thank you very much for your contribution! We plan to add sorting options on the logs page and use the default order on the chat page. Due to the need for design and front-end development, the merging of this PR might be slightly delayed.

Additionally, please check the CI errors.

**Comment by laipz8200** [comment]
You can run `dev/reformat` to fix the most lint errors automatically.

**Comment by leslie2046** [comment]
@laipz8200 
https://github.com/langgenius/dify/issues/7338
My issue will be resolved  if we can sort by create_at or update_at in logs page ,thank you!

File: api/services/conversation_service.py [hunk 1]
```diff
@@ -37,28 +39,30 @@ def pagination_by_last_id(cls, app_model: App, user: Optional[Union[Account, End
...
             if not last_conversation:
                 raise LastConversationNotExistsError()
 
-            conversations = base_query.filter(
-                Conversation.created_at < last_conversation.created_at,
-                Conversation.id != last_conversation.id
-            ).order_by(Conversation.created_at.desc()).limit(limit).all()
-        else:
-            conversations = base_query.order_by(Conversation.created_at.desc()).limit(limit).all()
+            # 根据排序方式构建过滤条件
```
Comment (laipz8200) [inline api/services/conversation_service.py]: ditto

File: api/services/conversation_service.py [hunk 2]
```diff
@@ -37,28 +39,30 @@ def pagination_by_last_id(cls, app_model: App, user: Optional[Union[Account, End
...
 
         has_more = False
         if len(conversations) == limit:
-            current_page_first_conversation = conversations[-1]
-            rest_count = base_query.filter(
-                Conversation.created_at < current_page_first_conversation.created_at,
-                Conversation.id != current_page_first_conversation.id
-            ).count()
+            current_page_last_conversation = conversations[-1]
+            # 构建剩余数量查询的过滤条件
```
Comment (laipz8200) [inline api/services/conversation_service.py]: ditto

File: api/services/conversation_service.py [hunk 3]
```diff
@@ -37,28 +39,30 @@ def pagination_by_last_id(cls, app_model: App, user: Optional[Union[Account, End
...
-                Conversation.created_at < last_conversation.created_at,
-                Conversation.id != last_conversation.id
-            ).order_by(Conversation.created_at.desc()).limit(limit).all()
-        else:
-            conversations = base_query.order_by(Conversation.created_at.desc()).limit(limit).all()
+            # 根据排序方式构建过滤条件
+            filter_condition = cls._build_filter_condition(sort_field, sort_direction, last_conversation)
+            base_query = base_query.filter(filter_condition)
+
+        # 应用排序
```
Comment (laipz8200) [inline api/services/conversation_service.py]: ditto

File: api/services/conversation_service.py [hunk 4]
```diff
@@ -37,28 +39,30 @@ def pagination_by_last_id(cls, app_model: App, user: Optional[Union[Account, End
         if exclude_ids is not None:
             base_query = base_query.filter(~Conversation.id.in_(exclude_ids))
 
-        if last_id:
-            last_conversation = base_query.filter(
-                Conversation.id == last_id,
-            ).first()
+        # 定义排序字段和方向
```
Comment (laipz8200) [inline api/services/conversation_service.py]: Please do not use Chinese comments in the code.

**Review by laipz8200 — CHANGES_REQUESTED** [review]

**Review by crazywoola — DISMISSED** [review]
@laipz8200

**Review by laipz8200 — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**